bluesky/log.py

```diff
@@ -125,22 +124,70 @@ def format(self, record):
         return formatted.replace("\n", "\n    ")
 
 
-plain_log_format = "[%(levelname)1.1s %(asctime)s.%(msecs)03d %(module)s:%(lineno)d] %(message)s"
+plain_log_format = "[%(levelname)1.1s %(asctime)s.%(msecs)03d %(module)15s:%(lineno)5d] %(message)s"
 color_log_format = ("%(color)s[%(levelname)1.1s %(asctime)s.%(msecs)03d "
-                    "%(module)s:%(lineno)d]%(end_color)s %(message)s")
+                    "%(module)15s:%(lineno)5d]%(end_color)s %(message)s")
+
+
+def color_logs(color):
+    """
+    If True, add colorful logging handler and ensure plain one is removed.
+
+    If False, do the opposite.
+    """
+    warnings.warn(f"The function color_logs is deprecated. "
+                  f"Use `config_bluesky_logging(color={color})` instead.")
+    config_bluesky_logging(color=color)
+
+
 logger = logging.getLogger('bluesky')
+doc_logger = logging.getLogger('bluesky.emit_document')
```

Comment:
Making public Logger objects is not necessary; they can just be accessed using `getLogger(...)` in the module where they are used.

Reply:
Pasting in from a private conversation on Slack, for the record:

That way the logging configuration module doesn't need to strictly depend on (import) the libraries it is doing logging for. For example, maybe beamlines sometimes use pyepics and sometimes use caproto. If I set up logging like:

```py
from pyepics import ...
from caproto import ...
```

I need to have both installed. If I set up logging like

```py
import logging
logging.getLogger('pyepics')
logging.getLogger('caproto')
```
then this will work even if pyepics or caproto are not installed.

Also, by setting up logging before the library is even imported, it ensures that we cannot accidentally miss any important log messages by setting up logging too late.

Reply:
Ab. So. Lute. Ly.  Setup logging before (almost) all else.

Reply:
Why was this marked as resolved, @ke-zhang-rd?

Reply:
Ah, I see it was removed from `__all__` in https://github.com/bluesky/bluesky/pull/1293/commits/4538846aae8f8d5e0f364a1e461f521a83e79f76. They will still show up in tab complete, but this is good enough for me.